hibernate-core/src/main/java/org/hibernate/engine/internal/JoinSequence.java

```diff
@@ -73,7 +73,7 @@ public JoinSequence getFromPart() {
 		return fromPart;
 	}
 
-	private Set<String> treatAsDeclarations;
+	protected Set<String> treatAsDeclarations;
```

Comment:
Why are you making this protected?  I'd much prefer a protected getter if we need protected access